src/chrome/content/scaffold/scaffold.js

```diff
@@ -303,6 +303,51 @@ var Scaffold = new function() {
 	});
 
 	/*
+	 * add template code
+	 */
+	this.addTemplate = Zotero.Promise.coroutine(function* (template, second) {
+		var cursorPos = _editors["code"].getSession().selection.getCursor();
+		var value = "";
+		switch(template) {
+			case "templateNewWeb":
+				value = Zotero.File.getContentsFromURL('chrome://scaffold/content/templates/newWeb.js');
+				break;
+			case "templateScrape":
+				switch(second) {
+					case "em":
+						value = Zotero.File.getContentsFromURL('chrome://scaffold/content/templates/scrapeEM.js');
+						break;
+					case "ris":
+						value = Zotero.File.getContentsFromURL('chrome://scaffold/content/templates/scrapeRIS.js');
+						break;
+					case "bibtex":
+						value = Zotero.File.getContentsFromURL('chrome://scaffold/content/templates/scrapeBIBTEX.js');
+						break;
+					case "marc":
+						value = Zotero.File.getContentsFromURL('chrome://scaffold/content/templates/scrapeMARC.js');
+						break;
+				}
+				break;
+			case "templateNewItem":
+				var fieldList = Zotero.ItemFields.getItemTypeFields(second)//new Zotero.Item("journalArticle");
+				for (var i=0; i<fieldList.length; i++) {
+					fieldList[i] = Zotero.ItemFields.getName(fieldList[i]);
+				}
+				document.getElementById('output').value = JSON.stringify(fieldList, null, '\t');
+				break;
+			case "templateAllTypes":
+				var types = Zotero.ItemTypes.getTypes();
+				var typesSimple = [];
+				for (var i=0; i<types.length; i++) {
+					typesSimple.push(types[i].name);
+				}
+				document.getElementById('output').value = JSON.stringify(typesSimple, null, '\t');
+				break;
+		}
```

Comment:
If you match `template` with the filenames the above can be simplified to:
```js
default:
    value = Zotero.File.getContentsFromURL(`chrome://scaffold/content/templates/${template}.js`);
    break
```
And there's probably no reason to complicate this function with multiple arguments even if in the UI there's distinction between new template and scrape templates.

Reply:
Very nice! I will do these changes.